rpc/src/rpc.rs

```diff
@@ -7355,6 +7514,269 @@ pub mod tests {
+                // --------------- Test Queries ---------------
+                // 1) Wallet1 - Owns 1 SPL Token
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    wallet1_pubkey,
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::SplTokenOwner).unwrap() == 1);
+                // 2) Wallet2 - Owns 2 SPL Tokens
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    wallet2_pubkey,
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::SplTokenOwner).unwrap() == 2);
+                // 3) Mint1 - Is in 2 SPL Accounts
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    mint1_pubkey,
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::SplTokenMint).unwrap() == 2);
+                // 4) Mint2 - Is in 1 SPL Account
+                let req = format!(
+                    r#"{{"jsonrpc":"2.0","id":1,"method":"getSecondaryIndexKeySize","params":["{}"]}}"#,
+                    mint2_pubkey,
+                );
+                let res = io.handle_request_sync(&req, meta.clone());
+                let result: Value = serde_json::from_str(&res.expect("actual response"))
+                    .expect("actual response deserialization");
+                let sizes: HashMap<RpcAccountIndex, usize> =
+                    serde_json::from_value(result["result"]["value"].clone()).unwrap();
+                assert!(*sizes.get(&RpcAccountIndex::SplTokenMint).unwrap() == 1);
+                // 5) SPL Token Program Owns 6 Accounts
```

Comment:
This is a bit shaky. Been trying to understand why the SPL Token Program is coming up with a size of 6. It owns the 3 token accounts I created, and 2 mint accounts, but I'm not sure where the 6th is coming from. Any Ideas?

Reply:
could dump them with `bank.get_program_accounts(...)`

Reply:
Ok will investigate more. Really hesitant to move on without understanding this discrepancy since this is a correctness check in a unit test.

Reply:
sry, by "dump" i meant query the bank for the programs with discrepancies and see what accounts are returned.  might help you debug

Reply:
Appears the extra token account is So11111111111111111111111111111111111111112. Seems like the SOL token account.

Token Program Accounts:

1) 1111111ogCyDbaRMvkdsHB3qfdyFYaG1WtRUAfdh: [lamports:111, owner:TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA, executable:false, rent_epoch:0]

2) 11111112D1oxKts8YPdTJRG5FzxTNpMtWmq8hkVx3: [lamports:333, owner:TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA, executable:false, rent_epoch:0]

3) 11111112cMQwSC9qirWGjZM6gLGwW69X22mqwLLGP: [lamports:444, owner:TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA, executable:false, rent_epoch:0]

4) 111111131h1vYVSYuKP6AhS86fbRdMw9XHiZAvAaj: [lamports:222, owner:TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA, executable:false, rent_epoch:0]

5) 11111113R2cuenjG5nFubqX9Wzuukdin2YfGQVzu5: [lamports:555, owner:TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA, executable:false, rent_epoch:0]

6) So11111111111111111111111111111111111111112: [lamports:1000000000, owner:TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA, executable:false, rent_epoch:1]